### PR TITLE
Frontend directory restructure

### PR DESCRIPTION
**Pull request by achen2401**
reference issue #13 

- moving `src` directory outside of the `static` directory, in effect, static directory will be empty from outset or in a in a fresh git checkout
- build process will: 
  (1) auto-populate the static directory with built JS files 
  (2) auto-generate the index.html which is then auto-populated in the templates directory 
  (3) a `dist`  directory will be created containing the built files ( `dist/js` containing js source files, `dist/templates` containing built index.html file)  ( (1) & (2) enable convenience for development) Please see webpack.js for details)
- added bug fixes - fix correct sorting by Last updated

@ivan-c Ivan, with this changes, the `Dockerfile` will need to be updated

### CODE REVIEW

File: dashboard/src/js/components/Profile.js [hunk 1]
```diff
@@ -45,6 +45,7 @@ function TabPanel(props) {
 }
 
 export default function ProfileDialog(props) {
+
```
Comment (pbugni) [inline dashboard/src/js/components/Profile.js]: extremely minor - but i don't think you intended this blank line addition?

**Review by pbugni — APPROVED** [review]

File: dashboard/src/js/components/Profile.js (hunk 1 above)
Comment (achen2401) [reply dashboard/src/js/components/Profile.js]: thank you, paul, for catching that, I removed the extra blank line.

**Review by ivan-c — APPROVED** [review]
Looks good!
I made some changes [in this follow-up PR](https://github.com/uwcirg/stayhome-dashboard/pull/16/) (merges into this one) that should allow us to only use `dist/`. 

After merging the #16 into this PR, you should be able to remove steps (1) and (2) to remove some redundancy (if that makes sense)


>(1) auto-populate the static directory with built JS files
>(2) auto-generate the index.html which is then auto-populated in the templates directory

**Comment by ivan-c** [comment]
Thanks for taking the time to try this out!

**Pull request merged**